modules/rgbd/src/colored_kinfu.cpp

```diff
@@ -344,9 +344,9 @@ void ColoredKinFuImpl<MatType>::render(OutputArray image, const Matx44f& _camera
 
 
 template< typename MatType >
-void ColoredKinFuImpl<MatType>::getCloud(OutputArray p, OutputArray n) const
+void ColoredKinFuImpl<MatType>::getCloud(OutputArray p, OutputArray n, OutputArray c) const
 {
-    volume->fetchPointsNormals(p, n);
+    dynamic_cast<ColoredTSDFVolume&>(*volume).fetchPointsNormalsColors(p, n, c);
```

Comment:
It will be better to add the function `fetchPointsNormalsColors` to Volume class.
Check [volume.hpp](https://github.com/opencv/opencv_contrib/blob/4.x/modules/rgbd/include/opencv2/rgbd/volume.hpp#L36-L39) and [colored_tsdf.cpp](https://github.com/opencv/opencv_contrib/blob/4.x/modules/rgbd/src/colored_tsdf.cpp#L73-L74) 
Shortly, add this function for all classes, and add `CV_Error` for not colored_tsdf classes

Reply:
changed. Note, that the previous code had similar effect, as the dynamic_cast would throw a std::bad_cast if the volume was not of the desired type.